test/e2e/config/e2e_conf.yaml

```diff
@@ -70,21 +70,41 @@ providers:
       targetName: "cluster-template-centos.yaml"
 
 variables:
-  KUBERNETES_VERSION: "${KUBERNETES_VERSION}"
-  UPGRADED_K8S_VERSION: "${UPGRADED_K8S_VERSION}"
```

Comment:
This declaration e2e_conf.yaml is meaningless since when using the e2e framework to read a file from e2e_config file it will check first if the var exist in variable environment in that case I will not even check the file so in this file we need to have only explicit value that can be overridden by variable environment.

Check the function workflow https://github.com/kubernetes-sigs/cluster-api/blob/3433f7b769b4e7f5cb899b2742a5a8a1a9f51b3e/test/framework/clusterctl/e2e_config.go#L553